File: HelpSoftware/Help/EF/Topic.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Help
{
  public class Topic
  {
    [Key]
    public int ID_Topic { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public Topic ParentTopic { get; set; }

    public List<Topic> ChildTopics { get; set; }

    public DateTime CreatedDate { get; set; }

    public User Author { get; set; }

    public User LastModifier { get; set; }

    public DateTime LastModified { get; set; }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/Topic.ts
import { User } from "./EF";

export class Topic {
  ID_Topic: Number;
  Name: string;
  Description: string;
  ParentTopic: Topic;
  ChildTopics: Topic[];
  CreatedDate: Date;
  Author: User;
  LastModifier: User;
  LastModified: Date;

  deserialize(input) {
    this.ID_Topic = input.ID_Topic;
    this.Name = input.Name;
    this.Description = input.Description;
    this.ParentTopic = input.ParentTopic;
    this.ChildTopics = input.ChildTopics;
    this.Author = new User().deserialize(input.Author);
    this.LastModifier = input.LastModifier;
    this.LastModified = input.LastModified;
    return this;
  }
}
